kaldb/src/main/java/com/slack/kaldb/logstore/search/LogIndexSearcherImpl.java

```diff
@@ -105,14 +105,14 @@ public SearchResult<LogMessage> search(
 
     Stopwatch elapsedTime = Stopwatch.createStarted();
     try {
-      Query query =
-          openSearchAdapter.buildQuery(dataset, queryStr, startTimeMsEpoch, endTimeMsEpoch);
-      span.tag("lucene_query", query.toString());
-
       // Acquire an index searcher from searcher manager.
       // This is a useful optimization for indexes that are static.
-
       IndexSearcher searcher = searcherManager.acquire();
+
+      Query query =
```

Comment:
Can we change this to
```

      try {
        Query query =
                openSearchAdapter.buildQuery(
                        dataset, queryStr, startTimeMsEpoch, endTimeMsEpoch, searcher);
        span.tag("lucene_query", query.toString());
      } catch (Exception e) {
        span.tag("lucene_query_parse_exception", "true");
      }
```

We'll be able to see in the future what queries were we unable to parse?

Reply:
These should already have exceptions for failure to parse, along with stack trace so I'm not sure having it on a trace is the best place for that? The overall trace should be marked as failed due to the exception as well.